### PR TITLE
Add nf-test for md5sum module

### LINKED ISSUE #5159: [FEATURE] nf-test for md5sum
### Is your feature request related to a problem? Please describe

port pytest to nf-test for md5sum module

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by human9**
Checks that the correct filename and contents are created. A custom config file is used to test extra args.

## PR checklist

Closes #5159 

- [X] `nf-test test modules/nf-core/md5sum/tests/main.nf.test --profile docker`

### CODE REVIEW

File: modules/nf-core/md5sum/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    tag "modules"
+    tag "modules_nfcore"
+    tag "md5sum"
+
+    test("md5sum on hello.txt") {
+
+        when {
+            process {
+                """
+                input[0] = Channel.of([
```
Comment (vagkaratzas) [inline modules/nf-core/md5sum/tests/main.nf.test:16]: is the Channel.of part needed here and on the next test? Never seen it in nf-tests

File: modules/nf-core/md5sum/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                    [ id: 'hello' ],
+                    [ file(params.modules_testdata_base_path + 'generic/txt/hello.txt', checkIfExists: true) ]
+                ])
+                """
+            }
+        }
+
+        then {
+            with(process.out) {
```
Comment (vagkaratzas) [inline modules/nf-core/md5sum/tests/main.nf.test:25]: assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
            
Should be working just fine for this module, no? (same for next test hopefully)
Also need a versions snapshot, and these two lines (if working properly) will cover that.

**Review by vagkaratzas — DISMISSED** [review]

File: modules/nf-core/md5sum/tests/main.nf.test (hunk 2 above)
Comment (human9) [reply modules/nf-core/md5sum/tests/main.nf.test]: You're right, that's much simpler! This was my first attempt to use nf-test so I may have gotten ahead of myself...

**Comment by human9** [comment]
Hmm, I don't understand why the conda test failed...

**Review by SPPearce — APPROVED** [review]

**Comment by SPPearce** [comment]
> Hmm, I don't understand why the conda test failed...

The conda test was failing because conda and docker/singularity were using different versions.
I've downgraded the conda version to match the ubuntu base (by copying what was used in another module).

**Pull request merged**